src/Workspaces/Core/Portable/EmbeddedLanguages/RegularExpressions/LanguageServices/RegexPatternDetector.cs

```diff
@@ -43,7 +43,7 @@ internal sealed class RegexPatternDetector
         /// Option names are the values from the <see cref="RegexOptions"/> enum.
         /// </summary>
         private static readonly Regex s_languageCommentDetector =
-            new Regex(@"\blang(uage)?\s*=\s*regex(p)?\b((\s*,\s*)(?<option>[a-zA-Z]+))*",
+            new Regex(@"^((//)|(')|(/\*))\s*\blang(uage)?\s*=\s*regex(p)?\b((\s*,\s*)(?<option>[a-zA-Z]+))*",
```

Comment:
`^` will only match the start of a comment.  So this means we won't search through a long comment to find a `lang=regex` match.  This means the .net regex engine will normally bail out after 3-4 characters.  i.e. once it sees `// <char>` it will normally be able to bail out.